cargo-concordium/src/build.rs

```diff
@@ -1265,6 +1269,72 @@ pub(crate) fn build_and_run_integration_tests(
     Ok(())
 }
 
+/// Runs a single unit test and prints out the result, returns an option so that
+/// succesful cases can be filtered
+fn get_test_result(
+    name: &Name,
+    seed: u64,
+    artifact: &Artifact<ArtifactNamedImport, CompiledFunction>,
+    enable_debug: bool,
+) -> Option<()> {
+    let mut is_err = false;
+    let test_name = name.as_ref().strip_prefix("concordium_test ")?;
+
+    // create a `TestHost` instance for each test with the usage flag set to `false`
+    let mut initial_state = trie::MutableState::initial_state();
+    let mut loader = trie::Loader::new(Vec::new());
+    let mut test_host = {
+        let inner = initial_state.get_inner(&mut loader);
+        let state = InstanceState::new(loader, inner);
+        TestHost::new(SmallRng::seed_from_u64(seed), state)
+    };
+
+    let test_result = artifact.run(&mut test_host, name, &[]).err().map(|msg| {
+        msg.downcast_ref::<ReportError>()
+            .cloned()
+            .unwrap_or_else(|| ReportError::Other {
+                msg: msg.to_string(),
+            })
+    });
+
+    match test_result {
+        Some(err) => {
+            is_err = true;
+            eprintln!(
+                "  - {} ... {}",
+                test_name,
+                Color::Red.bold().paint("FAILED")
+            );
+            eprintln!(
+                "    {} ... {}",
```

Comment:
Since this function is run in parallel, the different `eprintln!` might be intertwined right?
Maybe we should print these in sequence and only have the artifact.run be done in parallel.
What do you think?

Reply:
I see the issue, but won't moving this out of the parallelized section mean that the code will wait for all artifacts to finish before printing? I don't mind refactoring the implementation, but I would like to retain that functionality. I'll try something, until you come on again.

Reply:
How's this?

Reply:
:+1: Great approach